www/src/components/MarkdownPage/MarkdownPage.js

```diff
@@ -21,102 +21,146 @@ import StickyResponsiveSidebar from 'components/StickyResponsiveSidebar';
 import TitleAndMetaTags from 'components/TitleAndMetaTags';
 import findSectionForPath from 'utils/findSectionForPath';
 import toCommaSeparatedList from 'utils/toCommaSeparatedList';
-import {sharedStyles} from 'theme';
+import {media, sharedStyles} from 'theme';
 import {urlRoot} from 'constants';
 
-const MarkdownPage = ({
-  authors,
-  date,
-  ogDescription,
-  location,
-  markdownRemark,
-  sectionList,
-  titlePostfix = '',
-}) => {
-  const hasAuthors = authors.length > 0;
-  const titlePrefix = markdownRemark.frontmatter.title || '';
+class MarkdownPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.toggleMenuOverlay = this.toggleMenuOverlay.bind(this);
+    this.handleWindowResize = this.handleWindowResize.bind(this);
+    this.state = {
+      isMenuOverlayOpen: false,
+    };
+  }
+  componentDidUpdate(prevProps, prevState) {
+    if (this.state.isMenuOverlayOpen && !prevState.isMenuOverlayOpen) {
+      // Menu has been opened
+      window.addEventListener('resize', this.handleWindowResize);
+    }
+  }
+  handleWindowResize() {
+    const mediaQuery = media.greaterThan('small').replace('@media ', '');
+    if (window.matchMedia(mediaQuery).matches) {
+      // If the screen has been resized >= 'small'
+      this.toggleMenuOverlay(false);
+    }
+  }
+  toggleMenuOverlay(willOpen) {
+    this.setState(
+      {
+        isMenuOverlayOpen: willOpen,
+      },
+      () => {
+        this.context.onMenuOverlayToggle(this.state.isMenuOverlayOpen);
```

Comment:
Because Gatsby's layout can't re-render its children, this instead uses context to notify the layout to "sync" the header and footer visibility, based on whether than menu is open.

Reply:
> Because Gatsby's layout can't re-render its children

I believe you, and I'm also curious about this. Won't dig into it much for now but would be curious if you have time to comment more about that.

This seems a bit complicated to me. Going to read through it again and make sure I understand what is happening with this context call.

Reply:
I read through this again, and I'm still not sure why the `onMenuOverlayToggle` is not passed down via a prop instead of context? Then we could call `this.props.onMenuOverlayToggle` here. Context is fine too, but more obscure than just using props I think.

Reply:
I'd be keen on using props instead of context, for sure. I'm just not sure how this would be implemented? Would we pass arguments to `children()`?

Also, here's some notes about this setState callback architecture:

At the minute, Gatsby has its own sCU implementation in its route-level components. This means that, if we set some layout-level state, the route-level components (i.e. `<Docs>`) wouldn't re-render. So we have to work around this by keeping the state inside of `<Docs>`, and communicating back to the layout with context. (Which in turn has it's own "synced" state). Not ideal, but the best way I can think of atml

Reply:
I see - didn't realize this was a limitation of Gatsby, that we can't pass custom props to the children of the template. Makes sense!